### PR TITLE
Clarify that subsituted variables are treated as empty for the purpose of enclosing group rendering

### PR DESCRIPTION
**Pull request by bwiernik**
cf. https://discourse.citationstyles.org/t/groups-variables-and-missing-dates/1529/2 and citation-style-language/csl-evolution#9

Closes https://github.com/citation-style-language/csl-evolution/issues/9

### CODE REVIEW

**Comment by bwiernik** [comment]
Would also like @fbennett to weigh in here since he was involved in previous conversations here.

File: specification.rst [hunk 1]
```diff
@@ -1418,6 +1418,14 @@ empty. This accommodates descriptive ``cs:text`` elements. For example,
 can result in "retrieved from http://dx.doi.org/10.1128/AEM.02591-07", but
 doesn't generate output when the "URL" variable is empty.
 
+If a ``cs:group`` is nested within another ``cs:group``, the inner group is
+evaluated first: a non-empty nested ``cs:group`` is treated as a non-empty
+variable for the puropses of determining suppression of the outer ``cs:group``.
+
+If a ``cs:group`` contains a child ``cs:macro``, if the ``cs:macro`` is 
+non-empty, it is treated as a non-empty variable for the puropses of determining
```
Comment (denismaier) [inline specification.rst:1426]: puropses-> purposes

File: specification.rst [hunk 2]
```diff
@@ -1417,6 +1418,14 @@ empty. This accommodates descriptive ``cs:text`` elements. For example,
 can result in "retrieved from http://dx.doi.org/10.1128/AEM.02591-07", but
 doesn't generate output when the "URL" variable is empty.
 
+If a ``cs:group`` is nested within another ``cs:group``, the inner group is
+evaluated first: a non-empty nested ``cs:group`` is treated as a non-empty
+variable for the puropses of determining suppression of the outer ``cs:group``.
+
+If a ``cs:group`` contains a child ``cs:macro``, if the ``cs:macro`` is
```
Comment (bdarcus) [inline specification.rst:1425]:
```suggestion
When a ``cs:group`` contains a child ``cs:macro``, if the ``cs:macro`` is 
```

The two "if" words seemed odd.

**Comment by bwiernik** [comment]
@larsgw Do these address your concerns about ambiguity?

**Comment by larsgw** [comment]
LGTM, thank you.

**Comment by bwiernik** [comment]
@bdarcus RTM?

**Comment by bdarcus** [comment]
I'll check in the morning.

**Pull request merged**